File: agent-service/schemas/task.py
from typing import Literal, Optional, Dict, Any
from pydantic import BaseModel, Field
TaskStatus = Literal['created','planning','waiting_plan_approval','analyzing_codebase','developing_patch','reviewing','waiting_patch_approval','applying_patch','testing','fixing','completed','failed','cancelled','paused']
class TaskCreateRequest(BaseModel):
    workspaceRoot: str
    userRequest: str
    teamId: str = 'java-spring-team'
    workflowId: str = 'code-edit'
    mode: Literal['auto','semi-auto','manual'] = 'semi-auto'
    contextRefs: Dict[str, Any] = Field(default_factory=dict)
class TaskSummary(BaseModel):
    id: str
    status: TaskStatus
    title: str
    currentStep: Optional[str] = None

File: src/types/task.ts
export type TaskStatus =
  | 'created' | 'planning' | 'waiting_plan_approval' | 'analyzing_codebase' | 'developing_patch'
  | 'reviewing' | 'waiting_patch_approval' | 'applying_patch' | 'testing' | 'fixing'
  | 'completed' | 'failed' | 'cancelled' | 'paused';

export interface ContextRefs {
  currentFile?: boolean;
  selection?: boolean;
  terminalError?: boolean;
  gitDiff?: boolean;
  files?: string[];
}

export interface CreateTaskRequest {
  workspaceRoot: string;
  userRequest: string;
  teamId: string;
  workflowId: string;
  mode: 'auto' | 'semi-auto' | 'manual';
  targetAgent?: string;
  contextRefs: ContextRefs;
}

export interface TaskSummary {
  id: string;
  title: string;
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
  currentStep?: string;
}
